network/server/httpjson/handlers.go

```diff
@@ -108,23 +117,62 @@ func RegisterHandlers(mux *http.ServeMux, service interface{}, opts ServerOption
 		name := strings.ToLower(method.Name)
 		mux.HandleFunc(fmt.Sprintf("/%s", name), func(w http.ResponseWriter, r *http.Request) {
 			w.Header().Set("Content-Type", "application/json")
-			if strings.ToLower(r.Method) != "post" {
+
+			httpMethod := strings.ToUpper(r.Method)
+			if reqIn == nil && httpMethod != "GET" {
+				writeError(w, errRequestMustBeGet)
+				return
+			}
+			if reqIn != nil && httpMethod != "POST" {
 				writeError(w, errRequestMustBePost)
 				return
 			}
 
-			in := reflect.New(reqIn.Elem()).Interface()
-			defer r.Body.Close()
-			if err := json.NewDecoder(r.Body).Decode(in); err != nil {
-				writeError(w, errInvalidRequestBody)
-				return
+			headers := make(map[string]string)
+			for key, values := range r.Header {
+				if len(values) > 0 {
+					headers[key] = values[0]
+				}
 			}
 
-			svc := reflect.ValueOf(service)
+			var in interface{}
+			if reqIn != nil {
+				in = reflect.New(reqIn.Elem()).Interface()
+				defer r.Body.Close()
```

Comment:
Well yeah kind of..it's hidden in the comments about the `Body` field inside the `Request` definition:
```
        // Body is the request's body.
        //
        // For client requests a nil body means the request has no
        // body, such as a GET request. The HTTP Client's Transport
        // is responsible for calling the Close method.
        //
        // For server requests the Request Body is always non-nil
        // but will return EOF immediately when no body is present.
        // The Server will close the request body. The ServeHTTP
        // Handler does not need to.
        Body io.ReadCloser
```